perllib/FixMyStreet/Cobrand/BathNES.pm

```diff
@@ -0,0 +1,187 @@
+package FixMyStreet::Cobrand::BathNES;
+use parent 'FixMyStreet::Cobrand::Whitelabel';
+
+use strict;
+use warnings;
+
+use LWP::Simple;
+use URI;
+use Try::Tiny;
+use JSON::MaybeXS;
+
+sub council_area_id { return 2551; }
+sub council_area { return 'Bath and North East Somerset'; }
+sub council_name { return 'Bath and North East Somerset Council'; }
+sub council_url { return 'bathnes'; }
+
+sub contact_email {
+    my $self = shift;
+    return join( '@', 'councilconnect_rejections', 'bathnes.gov.uk' );
+}
+
+sub update_email {
+    my $self = shift;
+    return join( '@', 'highways', 'bathnes.gov.uk' );
+}
+
+sub map_type { 'BathNES' }
+
+sub example_places {
+    return ( 'BA1 1JQ', "Lansdown Grove" );
+}
+
+sub get_geocoder {
+    return 'OSM'; # default of Bing gives poor results, let's try overriding.
+}
+
+sub disambiguate_location {
+    my $self    = shift;
+    my $string  = shift;
+
+    my $town = 'Bath and North East Somerset';
+
+    # The council have provided a list of common typos which we should correct:
+    my %replacements = (
+        "broom" => "brougham",
+        "carnarvon" => "caernarvon",
+        "cornation" => "coronation",
+        "beafort" => "beaufort",
+        "beechan" => "beechen",
+        "malreword" => "malreward",
+        "canyerberry"=> "canterbury",
+        "clairemont"=> "claremont",
+        "salsbury"=> "salisbury",
+        "solsberry"=> "solsbury",
+        "lawn road" => "lorne",
+        "new road high littleton" => "danis house",
+    );
+
+    foreach my $original (keys %replacements) {
+        my $replacement = $replacements{$original};
+        $string =~ s/$original/$replacement/ig;
+    }
+
+    return {
+        %{ $self->SUPER::disambiguate_location() },
+        town   => $town,
+        centre => '51.3559192103294,-2.47522827137605',
+        span   => '0.166437921041471,0.429359043406088',
+        bounds => [ 51.2730478766607, -2.70792015294201, 51.4394857977022, -2.27856110953593 ],
+        string => $string,
+    };
+}
+
+sub pin_colour {
+    my ( $self, $p, $context ) = @_;
+    return 'grey' if $p->state eq 'not responsible';
+    return 'green' if $p->is_fixed || $p->is_closed;
+    return 'red' if $p->state eq 'confirmed';
+    return 'yellow';
+}
+
+sub send_questionnaires { 0 }
+
+sub enable_category_groups { 1 }
+
+sub default_show_name { 0 }
+
+sub default_map_zoom { 3 }
+
+sub map_js_extra {
+    my ($self, $c) = @_;
+
+    return unless $c->user_exists;
+
+    my $banes_user = $c->user->from_body && $c->user->from_body->areas->{$self->council_area_id};
+    if ( $banes_user || $c->user->is_superuser ) {
+        return ['/cobrands/bathnes/staff.js'];
+    }
+}
+
+sub category_extra_hidden {
+    my ($self, $meta) = @_;
+    my $code = $meta->{code};
+    return 1 if $code eq 'unitid' || $code eq 'asset_details';
```

Comment:
Did I see something in passing that said these were also used in non open311 fields? If so I wonder if it might be worth noting that in a comment?